compiler/ml/typecore.ml

```diff
@@ -3199,6 +3199,45 @@ and type_expect_ ?type_clash_context ?in_function ?(recarg = Rejected) env sexp
   | Pexp_extension ext ->
     raise (Error_forward (Builtin_attributes.error_of_extension ext))
   | Pexp_await _ -> (* should be handled earlier *) assert false
+  | Pexp_jsx_element
+      (Jsx_container_element
+         {
+           jsx_container_element_tag_name_start = tag_name;
+           jsx_container_element_props = props;
+         }) ->
+    let fields =
+      props
+      |> List.filter_map (function
+           | JSXPropValue (name, _, value) ->
+             Some
+               ( {
+                   txt = Longident.Ldot (Longident.Lident "JsxDOM", name.txt);
+                   loc = name.loc;
+                 },
+                 value,
+                 false )
+           | _ -> None)
+    in
+    let record = Ast_helper.Exp.record fields None in
```

Comment:
This idea here is to create a record of `JsxDOM.domProps` and type-check that.
It is of course laser focused to the `<div>` example.

(`children` could later be checked separately, as I believe we need to check if those `Jsx.element`)